rosapi/src/rosapi/params.py

```diff
@@ -93,11 +93,10 @@ def delete_param(name, params_glob):
         return
     # If the glob list is empty (i.e. false) or the parameter matches
     # one of the glob strings, continue to delete the parameter.
-    with param_server_lock:
-        if has_param(name):
+    if has_param(name, params_glob):
+        with param_server_lock:
```

Comment:
Isn't [L86-87](https://github.com/RobotWebTools/rosbridge_suite/pull/284/files#diff-9b18b072495ba40106802093bb486fb5R87) affected by the same issue?